core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -97,6 +98,11 @@ private[spark] object SizeEstimator extends Logging {
       return System.getProperty("spark.test.useCompressedOops").toBoolean
     }
 
+    // java.vm.info provides compressed ref info for IBM JDKs
+    if (System.getProperty("java.vendor").contains("IBM")) {
```

Comment:
Darn, right, we don't want to add that dependency. Well, I think for the moment these two changes are OK in that regard since adding a new dependency across these two isn't good.